src/Lean/Parser/Basic.lean

```diff
@@ -1444,7 +1444,7 @@ def longestMatchStep (left? : Option Syntax) (startSize startLhsPrec : Nat) (sta
     (s.pushSyntax successNode, prio) -- put successNode back on the stack
 
 def longestMatchMkResult (startSize : Nat) (s : ParserState) : ParserState :=
-  if !s.hasError && s.stackSize > startSize + 1 then s.mkNode choiceKind startSize else s
+  if s.stackSize > startSize + 1 then s.mkNode choiceKind startSize else s
```

Comment:
@leodemoura FYI, this change ensures that `longestMatch` returns exactly one node (assuming the nested parsers do as well) no matter what. Before that, the elaborator would elaborate *all* partial parses of an ill-formed command, which would lead to an ill-formed info tree that this PR incidentally uncovered.